### ISSUE TITLE
.HasAnyShortcode : return TRUE if a page uses at least one shortcode

### ISSUE THREAD

**Issue by jeremielp** [gohugoio/hugo#7996]
Similar to `.HasShortcode`, but without having the need to specify a shortcode name as parameter.

`.HasAnyShortcode` would Return TRUE if at least one shortcode is used in the page else Return FALSE

_Use case: CSS/JS libraries containing the code needed for a set of shortcodes. This would avoid to have to check for existence of all shortcodes one by one._

Alternatives could be:
- a specific parameter for `.HasShortcode`, like `.HasShortcode Any`
- use `.HasShortcode` with an array to test if one of the shortcodes in the array is used

**Comment by bep** [comment]
I agree about the "we need this", but I'm not sure we need another method. I think it fits with what we do in other areas of the template funcs that we have a vararg argument. So if we mage `HasShortCode(... string)` I think most people would understand if we document it.

/cc @moorereason

**Comment by jeremielp** [comment]
vararg argument fits my use case. I agree another method is not needed
Ideally, could be coupled with a `.Site.Shortcodes`, similar to `.Site.Taxonomies` / `.Site.Menus`, not to have to define / update the parameter each time a shortcode is added/removed

Or is there an other way to have access to the shortcode list that I did not found?

**Comment by bep** [comment]
OK, looking at the use case, I think I misunderstood the use case, but know I think. I do, and I take back what I said about "retrofitting" the `HasShortcode`. 

I think we need a method `HasShortCodeAny` which returns true if one of the given shortcode names is in use.

**Comment by moorereason** [comment]
First, as is usually the case, when I program late at night, I usually missing something.  In this case, I didn't catch the fact that this PR is proposing two changes:

1. Allow to test for any unnamed shortcodes.
1. Allow to test for any of a list of named shortcodes.

My PR retrofitting `HasShortcode` only attempted to address the first request.

@bep,
As for your new proposal, are you suggesting we leave `HasShortcode` unchanged and put these new features into `HasShortcodeAny` alone?

**Comment by bep** [comment]
Let us create `HasShortcodeAny`; which returns true _is any of the shortcode names given is in use_.

**Comment by moorereason** [comment]
What about the existing PR and the *test for any unnamed shortcodes* case?

**Comment by jeremielp** [comment]
FYI, what I finally did, as I had the case for both partials and shortcodes, is setting a boolean to true in the partial/shortcode using .Scratch.Set, and checking for that boolean to include or not my javascripts. These are all in the footer, after the shortcodes. I have not tested if this work for CSS in the <head> though.
